WalletWasabi/BitcoinCore/Mempool/MempoolMirror.cs

```diff
@@ -127,7 +127,10 @@ public IEnumerable<Transaction> GetSpenderTransactions(IEnumerable<OutPoint> txO
 				{
 					if (mempoolTx.Value.Inputs.Select(x => x.PrevOut).Contains(input))
 					{
-						spenders.Add(mempoolTx.Key, mempoolTx.Value);
+						if (!spenders.TryAdd(mempoolTx.Key, mempoolTx.Value))
+						{
+							Logger.LogInfo("Transaction hash has already been added.");
```

Comment:
I don't get the need of this log entry, to me it is completely unnecessary. I would take the opportunity to improve this algorithm to make it a bit faster and more functional. Here my suggestion:

```c#
		var mempool = Mempool.Values;
		var txOutSet = txOuts.ToHashSet();
		return mempool
			.SelectMany(tx => tx.Inputs.Select(i => (tx, i.PrevOut)))
			.Where(x => txOutSet.Contains(x.PrevOut))
			.Select(x => x.tx);
```

Reply:
Yes, the log is unnecessary. It just felt wrong to leave it empty.